bundle/jsky.app.ot/src/main/scala/jsky/app/ot/ags/BagsManager.scala

```diff
@@ -39,54 +42,33 @@ object BagsManager {
       BagsTask(observation, curCtx, ctxOpt)
 
     def performLookup(): Unit = {
-      LOG.info(s"Performing BAGS lookup for observation=${observation.getObservationID}")
-
-      def resultString[A](r: Option[A]): String =
-        s"Results=${r ? "Yes" | "No"}."
-
-      // We must handle GeMS separately from other strategies, and delegate the work to the GemsGuideStarWorker.
-      def gemsLookup(obsCtx: ObsContext): Unit = {
-        LOG.info(s"BAGS GeMS lookup for observation=${observation.getObservationID} starting...")
-        Try { GemsGuideStarWorker.findGuideStars(obsCtx) } match {
-          case Success(ggs) =>
-            LOG.info(s"BAGS GeMS lookup for observation=${observation.getObservationID} successful. ${resultString(ggs.asScalaOpt)}")
-            Swing.onEDT {
-              muteObservation(observation)
-              GemsGuideStarWorker.applyResults(TpeContext(observation), ggs, true)
-              unmuteObservation(observation)
-            }
-            taskComplete(this, success = true)
-          case Failure(ex) =>
-            LOG.warning(s"BAGS GeMS lookup for observation=${observation.getObservationID} failed. Exception=$ex")
-            taskComplete(this, success = false)
-        }
-      }
-
-      // The mechanism to perform lookups for all other guide probes.
-      def otherLookup(obsCtx: ObsContext, strategy: AgsStrategy): Unit = {
-        val fut = strategy.select(obsCtx, OT.getMagnitudeTable)
-        fut.onComplete {
+      def lookup[S,T](optExtract: S => Option[T])(worker: (TpeContext, S) => Unit)(results: Try[S]): Unit = {
+        results match {
           case Success(selOpt) =>
-            LOG.info(s"BAGS lookup for observation=${observation.getObservationID} successful. ${resultString(selOpt)}")
-            Swing.onEDT {
-              muteObservation(observation)
-              GuideStarWorker.applyResults(TpeContext(observation), selOpt)
-              unmuteObservation(observation)
+            LOG.info(s"BAGS lookup for observation=${observation.getObservationID} successful. Results=${optExtract(selOpt) ? "Yes" | "No"}.")
+            if (ObservationStatus.computeFor(observation) != ObservationStatus.OBSERVED) {
+              Swing.onEDT {
+                muteObservation(observation)
+                worker(TpeContext(observation), selOpt)
+                unmuteObservation(observation)
+              }
             }
             taskComplete(this, success = true)
+
           case Failure(ex) =>
             LOG.warning(s"BAGS lookup for observation=${observation.getObservationID} failed. Exception=$ex")
```

Comment:
I don't mind at all. My only concern is that this log message is output if BAGS is attempting to run and the network is out, which could get very verbose.

Perhaps I should match on the exception and output a separate message if that is the case. The only problem I see with this is that a `GenericError` is what is reported if the catalog host is unreachable, and from what I can tell, this may be issued in other circumstances.

For now, for a `GenericError`, I will simply output the message, and for any other error type, I will output the stack trace.